### PR TITLE
[CELEBORN-1266] Improve log of current failed workers for WorkerStatusTracker

### PR DESCRIPTION
**Pull request by SteNicholas**
### What changes were proposed in this pull request?

Improve log of current failed workers for `WorkerStatusTracker#recordWorkerFailure` and `WorkerStatusTracker#handleHeartbeatResponse`.

### Why are the changes needed?

It's recommended to improve the log of current failed workers in `recordWorkerFailure` and `handleHeartbeatResponse` of `WorkerStatusTracker`. Meanwhile the log level of current failed workers could be warn.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

No.

### CODE REVIEW

**Comment by SteNicholas** [comment]
@FMX, @AngersZhuuuu, could you help to review the log improvement of `WorkerStatusTracker`?

File: master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala [hunk 1]
```diff
@@ -978,7 +978,8 @@ private[celeborn] class Master(
     if (shouldResponse) {
       context.reply(HeartbeatFromApplicationResponse(
         StatusCode.SUCCESS,
-        new util.ArrayList(statusSystem.excludedWorkers),
+        new util.ArrayList(
+          (statusSystem.excludedWorkers.asScala ++ statusSystem.manuallyExcludedWorkers.asScala).asJava),
```
Comment (AngersZhuuuu) [inline master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala:982]: Another PR?

**Review by AngersZhuuuu — APPROVED** [review]
LGTM

**Review by AngersZhuuuu — APPROVED** [review]

**Comment by SteNicholas** [comment]
Merged into main(v0.5.0) and branch-0.4(v0.4.1).

**Pull request closed without merging**